bluemira/equilibria/coils/_grouping.py

```diff
@@ -0,0 +1,1088 @@
+
+        Parameters
+        ----------
+        output: np.ndarray
+            Output of calculation
+        sum_coils: bool
+            sum over coils
+        control: bool
+            operations on control coils only
+
+        Returns
+        -------
+        np.ndarray
+            Response output
+        """
+        ind = self._control_ind if control else slice(None)
+
+        return np.sum(output[..., ind], axis=-1) if sum_coils else output[..., ind]
+
+    def get_coiltype(self, ctype):
+        """Get coils by coils type"""
+        return CoilSet(*super()._get_coiltype(ctype))
+
+    @property
+    def area(self) -> float:
+        """
+        Cross sectional area of CoilSet
+        """
+        return np.sum(super().area)
+
+    @property
+    def volume(self) -> float:
+        """
+        Volume of Coilset
+        """
+        return np.sum(super().volume)
+
+    def psi(
+        self, x: np.ndarray, z: np.ndarray, sum_coils: bool = True, control: bool = False
+    ) -> np.ndarray:
+        """
+        Psi of Coilset
+
+        Parameters
+        ----------
+        x: np.ndarray
+            The x values at which to calculate the psi response
+        z: np.ndarray
+            The z values at which to calculate the psi response
+        sum_coils: bool
+            sum over coils
+        control: bool
+            operations on control coils only
+
+        Returns
+        -------
+        np.ndarray of response
+        """
+        return self._sum(super().psi(x, z), sum_coils=sum_coils, control=control)
+
+    def Bx(
+        self, x: np.ndarray, z: np.ndarray, sum_coils: bool = True, control: bool = False
+    ) -> np.ndarray:
+        """
+        Bx of Coilset
+
+        Parameters
+        ----------
+        x: np.ndarray
+            The x values at which to calculate the Bx response
+        z: np.ndarray
+            The z values at which to calculate the Bx response
+        sum_coils: bool
+            sum over coils
+        control: bool
+            operations on control coils only
+
+        Returns
+        -------
+        np.ndarray of response
+        """
+        return self._sum(super().Bx(x, z), sum_coils=sum_coils, control=control)
+
+    def Bz(
+        self, x: np.ndarray, z: np.ndarray, sum_coils: bool = True, control: bool = False
+    ) -> np.ndarray:
+        """
+        Bz of Coilset
+
+        Parameters
+        ----------
+        x: np.ndarray
+            The x values at which to calculate the Bz response
+        z: np.ndarray
+            The z values at which to calculate the Bz response
+        sum_coils: bool
+            sum over coils
+        control: bool
+            operations on control coils only
+
+        Returns
+        -------
+        np.ndarray of response
+        """
+        return self._sum(super().Bz(x, z), sum_coils=sum_coils, control=control)
+
+    def unit_psi(
```

Comment:
Not really a fan of `unit_..` as a name for these methods. The whole point is to get the response at a location for a unit _current_ in a given coil or grouping thereof for [psi, Bx, Bz, Bp]. 

As such, e.g. `unit_psi(x, z)` is effectively doing `get_response_to_a_unit_current_in_coil_for_psi_at_location(x, z)`. I feel that the previous formulation of `control_psi(x, z)` or e.g. `get_psi_response(x, z)` are clearer short-forms for this.

The `unit` aspect is perhaps the least interesting part of this, and is more readily confused with e.g. [V.s], [Wb/rad], etc.

Reply:
control then is confused with the concept of control coils which is different. I could change the to `<>_response` possibly. Get makes me think that there should be a set